src/components/Cart/Cart.test.js

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+import Cart from './Cart';
+
+describe('Cart', () => {
+  describe('renders Cart visible only when isCartVisible state is true', () => {
+    it('applies class "invisible" to Cart when isCartVisible is false', () => {
```

Comment:
This description of the locks you in with the current implementation. Test descriptions should not contain implementation details like variable names and class `invisible`.  You should not check for the existence of a class, Jest has special methods for checking this https://stackoverflow.com/questions/67744440/jest-and-react-testing-library-how-to-test-if-an-element-is-hidden. Sometimes your test will contain implementation details, but when possible you should try to avoid it.

Reply:
Yep, that's probably right, you can find [here](https://github.com/testing-library/jest-dom#tobevisible) how `toBeVisible` works https://github.com/testing-library/jest-dom#tobevisible . Indeed, you cannot animate the `display` property. You can animate `opacity`. So in conclusion for this situation if the scaling animation is required, your approach to test for the `invisible` class is the best thing you can do.